### ISSUE TITLE
[Discussion] Partners, Competition - do we need all this?

### ISSUE THREAD

**Issue by nikdejan** [YetiForceCompany/YetiForceCRM#2534]
There is no documentation so I'm not clear that I clearly understand YF Team intentions with this modules, but at seams to me that we do not need them all.
First, partners, vendors, competition, customers - all this entities should be single module, because all this entities will probably have the the same fields (system fields, legal form, address etc). Instead of creating separate module for partners, competition etc, it is enough to have one type picklist (multiselect preferably) with values Partner, Competition etc. If you need some specific fields you can relate blocks of fields with picklist so different blocks will appear depending on picklist value.
Second problem is that the same company can be your partner and/or customer and/or competition. How would you handle this situation if you have separate modules for each of this relations? 
I understand the need to separate partners from competition etc, but better way to do that is by adding filters in different situations inside crm. For example, if you have reference field Partner, when you click Search icon in order to select from existing Accounts, you will get a filtered list of Accounts based on the value of multiselect picklist which I mentioned previously. This means that 1:M field will have additional attribute which will define filter criteria. This will help us have smaller number modules but also will open up many new possibilities.

**Comment by eptapsyxos** [comment]
The same questions come uped in my mind when i saw the new modules.i still don't understand the usage.

**Comment by ChristopheLux** [comment]
Hello I clearly go with nikdejan on this one it makes so much more sense. You can have multiple type of relations with one company or even an individual...

**Comment by witer30** [comment]
and I like separate modules for such information...
Only what would be nice to have is possibility to convert first entry to Partner or Competitor id there is multi role to avoid entering data separately in every module... 
if you have one module and multirole how you can manage this? additionally this can be confusing for example during some report generation because you have to remember to always add role as a filter... 
Completly follow YF...

**Comment by nikdejan** [comment]
@witer30: "you have to remember to always add role as a filter" Come on! This is like you said you have to remember to choose the right module. 
"multi role to avoid entering data separately in every module... if you have one module and multirole how you can manage this?" You enter data only once in a single module. With converting entries between multiple module you are (kind of) solving the problem of multiple data entering. With single module, you DON'T EVEN HAVE THAT PROBLEM.
Software should reflect real world as close as possible. You have Company A. It is unnatural to have more than one entry of Company A in your system. Company A as a customer is not different from Company A as a partner. Relations are multiple, but the company is one and the same. 
Suppose you want to schedule an Activity related to Company A. To which Company A are you going to relate it? You receive a phone call from Company A. To which Company A are you going to relate it? To Company-A-as-a-Customer, or to Company-A-as-a-Partner? You receive an email. To which Company A it is going to be attached? And so on. 
I suggest multiselect picklist to contain type of relation to an account over multiple modules. But even this picklist might not be needed in certain scenarios. You don't even have to decide if account is customer and/or partner and/or competition. You simply enter account details and system will understand your business relation (customer, partner etc) to that account based on records related to that account. For example, if you have Sales Orders related to particular account, then it is obvious that this account is your customer.
Last time I installed YetiForce it had about 800 tables in a database. Now we are going to have even more. Without getting into detail, based on number of tables only, one could ask if this is really a good design.
If YF want's to be the best crm ever, then creating hundreds of modules which are almost identical is not a right direction. You do not need to predict every possible scenario and then create a module to cover it. Instead, you should create a system that is customizable enough to cover every need.
Way to power is not towards quantity but towards quality - not towards concretisation and particularisation but towards abstraction and generalisation.

**Comment by reeid** [comment]
Clear enough mate. Such solutions is the same category as "Our sites" Is there anyone who use it?!
Another story is Payments out and Payments in. Why separate this records in two tables???? 
Advanced system of setting privileges is not going together with this few introduced modules.
.

**Comment by bpabiszczak** [comment]
1. If there's a module you don't need – just disable it :]
2. Even if they'd have common basic fields, there are still so many differences between these modules that we decided to divide them [add new ones].
3. If someone is both, a customer and a vendor, then he should be added in both modules [it works like this in SAP and many other systems].

We have ambitious plans for each of these modules separately.

**Comment by websafe** [comment]
@bpabiszczak This is somehow strange but OK - I can disable all modules except Accounts and live with that. But what I don't understand is the way contacts are assigned - assuming I'm using for example Accounts, Vendors, Partners . When I have a company, which needs to be in Accounts and Vendors and Partners do I have to add all Contacts in that company **three times**? 1) assigned to the Account, 2) assigned to the Vendor, 3) assigned to a Partner? How to keep them all up to date? Or is there another way to have one Contact assigned to an Account, a Partner, and a Vendor at once?
